### PR TITLE
add doc generator

### PR DESCRIPTION
**Pull request by edward-io**
Summary:
PyTorch style docs using Sphinx.

image

Test plan:
`cd docs`
`make html`

### CODE REVIEW

File: docs/source/index.rst [hunk 1]
```diff
@@ -0,0 +1,43 @@
+.. TorchRec documentation master file, created by
+   sphinx-quickstart on Fri Jan 14 11:37:03 2022.
+   You can adapt this file completely to your liking, but it should at least
+   contain the root `toctree` directive.
+
+Welcome to the TorchSnapshot documentation!
+===========================================
+
+TorchRec is a PyTorch library for adding fault tolerance to large-scale PyTorch distributed training workloads.
```
Comment (ananthsub) [inline docs/source/index.rst:9]:
```suggestion
TorchSnapshot is a PyTorch library for adding fault tolerance to large-scale PyTorch distributed training workloads.
```

File: docs/source/conf.py [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+# This pattern also affects html_static_path and html_extra_path.
+exclude_patterns = []
+
+
+# -- Options for HTML output -------------------------------------------------
+
+# The theme to use for HTML and HTML Help pages.  See the documentation for
+# a list of builtin themes.
+#
+# html_theme = 'alabaster'
```
Comment (ananthsub) [inline docs/source/conf.py:61]: Not needed?

File: docs/source/conf.py [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+print(target_dir)
+
+# -- Project information -----------------------------------------------------
+
+project = "TorchSnapshot"
+copyright = "2022, Meta"
+author = "Meta"
+
+# The full version, including alpha/beta/rc tags
+release = "0.0.1"
```
Comment (ananthsub) [inline docs/source/conf.py:37]: Should we use the version from the source?

File: docs/source/index.rst [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+
+`Installation instructions <https://github.com/pytorch/torchsnapshot#readme>`_
+
+Examples
+--------
+
+* `Simple example <https://github.com/pytorch/torchsnapshot/blob/main/examples/simple_example.py>`_
+* `Using snapshot with distributed data parallel (DDP) <https://github.com/pytorch/torchsnapshot/blob/main/examples/ddp_example.py>`_
+* `Integration with TorchRec example <https://github.com/pytorch/torchsnapshot/blob/main/examples/torchrec_example.py>`_
```
Comment (ananthsub) [inline docs/source/index.rst:18]:
```suggestion
* `Using TorchSnapshot with TorchRec <https://github.com/pytorch/torchsnapshot/blob/main/examples/torchrec_example.py>`_
```

File: docs/source/index.rst [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+
+TorchRec is a PyTorch library for adding fault tolerance to large-scale PyTorch distributed training workloads.
+
+`Installation instructions <https://github.com/pytorch/torchsnapshot#readme>`_
+
+Examples
+--------
+
+* `Simple example <https://github.com/pytorch/torchsnapshot/blob/main/examples/simple_example.py>`_
+* `Using snapshot with distributed data parallel (DDP) <https://github.com/pytorch/torchsnapshot/blob/main/examples/ddp_example.py>`_
```
Comment (ananthsub) [inline docs/source/index.rst:17]:
```suggestion
* `Using TorchSnapshot with DistributedDataParallel (DDP) <https://github.com/pytorch/torchsnapshot/blob/main/examples/ddp_example.py>`_
```

File: torchsnapshot/__init__.py [hunk 6]
```diff
@@ -7,10 +7,10 @@
 
 "A lightweight library for adding fault tolerance to large-scale PyTorch distributed training workloads"
 
-from .rng_state import RNGState
 from .snapshot import Snapshot
-from .state_dict import StateDict
 from .stateful import Stateful
+from .state_dict import StateDict
+from .rng_state import RNGState
```
Comment (ananthsub) [inline torchsnapshot/__init__.py:13]: This is now out of order?

**Review by ananthsub — COMMENTED** [review]
Exciting!

File: torchsnapshot/__init__.py (hunk 6 above)
Comment (edward-io) [reply torchsnapshot/__init__.py:13]: I thought reordering this would change the ordering in the docs, but it doesn't. Will revert

**Comment by edward-io** [comment]
cd docs; make html to build docs
to preview: cd build/html; python -m http.server
If using a server that isn't accessible from the internet (i.e. AWS or devserver), use SSH tunneling
ssh my-host-name-here -L 1234:localhost:1234
keep this connection open
python -m http.server 1234
Access from your web browser: http://localhost:1234/

**Pull request closed without merging**